src/zenml/pipelines/base_pipeline.py

```diff
@@ -92,42 +92,92 @@ def __init__(self, *args: BaseStep, **kwargs: BaseStep) -> None:
 
         self.enable_cache = getattr(self, PARAM_ENABLE_CACHE)
         self.pipeline_name = self.__class__.__name__
-        self.__steps = dict()
         logger.info(f"Creating pipeline: {self.pipeline_name}")
         logger.info(
             f'Cache {"enabled" if self.enable_cache else "disabled"} for '
             f"pipeline `{self.pipeline_name}`"
         )
 
-        if args:
+        self.__steps: Dict[str, BaseStep] = {}
+        self._verify_arguments(*args, **kwargs)
+
+    def _verify_arguments(self, *steps: BaseStep, **kw_steps: BaseStep) -> None:
+        """Verifies the initialization args and kwargs of this pipeline.
+
+        This method makes sure that no missing/unexpected arguments or
+        arguments of a wrong type are passed when creating a pipeline. If
+        all arguments are correct, saves the steps to `self.__steps`.
+
+        Args:
+            *steps: The args passed to the init method of this pipeline.
+            **kw_steps: The kwargs passed to the init method of this pipeline.
+
+        Raises:
+            PipelineInterfaceError: If there are too many/few arguments or
+                arguments with a wrong name/type.
+        """
+        input_step_keys = list(self.STEP_SPEC.keys())
+        if len(steps) > len(input_step_keys):
             raise PipelineInterfaceError(
-                "You can only use keyword arguments while you are creating an "
-                "instance of a pipeline."
+                f"Too many input steps for pipeline '{self.pipeline_name}'. "
```

Comment:
We can also show what was passed in i.e. `input_step_keys` and what is expected `steps`. Or even the `set(input_step_keys) - set(steps)`?

Reply:
So you would include the expected step keys as well as the *args and **kwargs in the error message? Not sure if that might not get a little overwhelming in the error message then.

This error is actually just if *args is already longer than the expected amount of steps, so we can't really log the keys which are too much unfortunately.

Reply:
Oh if its just *args then lets leave it like this :-)